service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/ainotebook/post-startup.sh

```diff
@@ -756,6 +757,26 @@ fi
 chown ${LOGIN_USER}:${LOGIN_USER} "${USER_BASHRC}"
 chown ${LOGIN_USER}:${LOGIN_USER} "${USER_BASH_PROFILE}"
 
+####################################
+# Run a user provided startup script
+####################################
+
+# If the user has provided a startup script, run it after workbench setup,
+# but before restarting Jupyter and running tests.
+
+readonly USER_STARTUP_SCRIPT="$(get_metadata_value "instance/attributes/terra-user-startup-script")"
+if [[ -n "${USER_STARTUP_SCRIPT}" ]]; then
+  emit "Running user provided startup script..."
+  readonly USER_STARTUP_SCRIPT_FILE="${USER_TERRA_CONFIG_DIR}/user-startup-script.sh"
+
+  # Copy the user's startup script to the user's .terra directory
+  curl -L "${USER_STARTUP_SCRIPT}" > "${USER_STARTUP_SCRIPT_FILE}"
+
+  # Run the user's startup script as the root user so that they made install packages
+  chmod +x "${USER_STARTUP_SCRIPT_FILE}"
+  source "${USER_STARTUP_SCRIPT_FILE}"
```

Comment:
I don't think we want to source the script as that can produce side effects for this script.
I think we just want to execute it.

We should also log output from the script to a known location, such as `~/.terra/user-startup-output.txt` (too bad our log file is ".txt" instead of ".log").

Would be good to update the above to:

```
  emit "Running user provided startup script, output to ${USER_STARTUP_OUTPUT_FILE}..."
```